tezos/new_context/src/tezedge_context.rs

```diff
@@ -57,25 +58,24 @@ impl TezedgeIndex {
         let patch_context = Rc::new(patch_context);
         Self {
             patch_context,
-            strings: Default::default(),
             repository,
+            storage: Default::default(),
         }
     }
 
-    pub fn get_str(&self, s: &str) -> Rc<str> {
-        let mut strings = self.strings.borrow_mut();
-        strings.get_str(s)
-    }
-
     pub fn find_entry_bytes(&self, hash: HashId) -> Result<Option<Vec<u8>>, DBError> {
         let repo = self.repository.read()?;
         Ok(repo.get_value(hash)?.map(|v| v.to_vec()))
     }
 
-    pub fn find_entry(&self, hash: HashId) -> Result<Option<Entry>, DBError> {
+    pub fn find_entry(
+        &self,
+        hash: HashId,
+        storage: &mut Storage,
```

Comment:
Can you explain why this `&mut Storage` argument is needed? it is part of `self`, so I guess the reason is that it needs to be `&mut`, but why is that?